packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -14,221 +14,195 @@
+    AttributeKeyValue.doubleValue = value;
   }
 
-  return attributeValue;
+  AttributeKeyValue.type = aType;
+
+  return AttributeKeyValue;
 }
 
 /**
- * convert events
+ *
+ * convert events to proto
  * @param events array of events
- * @param maxAttributes - maximum number of event attributes to be converted
  */
 export function toCollectorEvents(
-  events: TimedEvent[]
-): collectorTypes.TimeEvents {
-  let droppedAnnotationsCount = 0;
-  let droppedMessageEventsCount = 0; // not counting yet as messageEvent is not implemented
-
-  const timeEvent: collectorTypes.TimeEvent[] = events.map(
-    (event: TimedEvent) => {
-      let attributes: collectorTypes.Attributes | undefined;
-
-      if (event && event.attributes) {
-        attributes = toCollectorAttributes(event.attributes);
-        droppedAnnotationsCount += attributes.droppedAttributesCount || 0;
-      }
-
-      let annotation: collectorTypes.Annotation = {};
-      if (event.name || attributes) {
-        annotation = {};
-      }
-
-      if (event.name) {
-        annotation.description = toCollectorTruncatableString(event.name);
-      }
-
-      if (typeof attributes !== 'undefined') {
-        annotation.attributes = attributes;
-      }
-
-      // @TODO convert from event.attributes into appropriate MessageEvent
-      // const messageEvent: collectorTypes.MessageEvent;
-
-      const timeEvent: collectorTypes.TimeEvent = {
-        time: hrTimeToTimeStamp(event.time),
-        // messageEvent,
-      };
-
-      if (annotation) {
-        timeEvent.annotation = annotation;
-      }
-
-      return timeEvent;
-    }
-  );
-
-  return {
-    timeEvent,
-    droppedAnnotationsCount,
-    droppedMessageEventsCount,
-  };
-}
-
-/**
- * determines the type of link, only parent link type can be determined now
- * @TODO refactor this once such data is directly available from {@link Link}
- * @param span
- * @param link
- */
-export function toCollectorLinkType(
-  span: ReadableSpan,
-  link: Link
-): collectorTypes.LinkType {
-  const linkSpanId = link.context.spanId;
-  const linkTraceId = link.context.traceId;
-  const spanParentId = span.parentSpanId;
-  const spanTraceId = span.spanContext.traceId;
-
-  if (linkSpanId === spanParentId && linkTraceId === spanTraceId) {
-    return collectorTypes.LinkType.PARENT_LINKED_SPAN;
-  }
-  return collectorTypes.LinkType.UNSPECIFIED;
-}
-
-/**
- * converts span links
- * @param span
- */
-export function toCollectorLinks(span: ReadableSpan): collectorTypes.Links {
-  const collectorLinks: collectorTypes.Link[] = span.links.map((link: Link) => {
-    const collectorLink: collectorTypes.Link = {
-      traceId: hexToBase64(link.context.traceId),
-      spanId: hexToBase64(link.context.spanId),
-      type: toCollectorLinkType(span, link),
+  timedEvents: TimedEvent[]
+): opentelemetryProto.trace.v1.Span.Event[] {
+  const protoEvents: opentelemetryProto.trace.v1.Span.Event[] = [];
+  timedEvents.forEach(timedEvent => {
```

Comment:
I would prefer to use `timedEvents.map` instead of `timedEvents.forEach`. Similar to `toCollectorLinks` function.